tools/code-generation/generator/src/main/java/com/amazonaws/util/awsclientgenerator/domainmodels/codegeneration/ServiceModel.java

```diff
@@ -61,6 +61,16 @@ public boolean hasOnlyBearerAuth() {
         return operations.values().parallelStream().allMatch(operation -> operation.getSignerName().equals("Aws::Auth::BEARER_SIGNER"));
     }
 
+    public Operation getOperationForRequestShapeName(String requestShapeName) {
+        for (Map.Entry<String, Operation> opEntry : operations.entrySet()) {
+            Operation op = opEntry.getValue();
+            if (op.getRequest() != null && op.getRequest().getShape().getName() == requestShapeName) {
+                return op;
+            }
+        }
+        return null;
```

Comment:
shouldnt ever return nullptr in java, should return [optional](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html) and deal with presence or absence of optional.